src/server/pps/server/api_server.go

```diff
@@ -511,62 +511,79 @@ func (a *apiServer) ListDatum(ctx context.Context, request *pps.ListDatumRequest
 		}
 		return datumHash, nil
 	}
+	for _, fileInfo := range allFileInfos.FileInfo {
+		_, err := pathToDatumHash(fileInfo.File.Path)
```

Comment:
Hmm ... not sure what you mean. I need to remove some of the file infos before I sort them, and that is done whether or not the response is paginated.